gui/mozregui/report.py

```diff
@@ -60,11 +61,33 @@ def status_text(self):
         return msg % self.state_text
 
 
+def bulk_action_slots(slots, signal_object, slot_object, action):
+    if signal_object and slot_object:
+        for name in slots:
+            signal = getattr(signal_object, name)
+            slot = getattr(slot_object, name)
+            if action == "connect":
+                signal.connect(slot)
+            elif action == "disconnect":
+                signal.disconnect(slot)
+            else:
+                assert False
+
+
+def bulk_connect_slots(slots, signal_object, slot_object):
+    bulk_action_slots(slots, signal_object, slot_object, 'connect')
+
+
+def bulk_disconnect_slots(slots, signal_object, slot_object):
+    bulk_action_slots(slots, signal_object, slot_object, 'disconnect')
+
+
 class ReportModel(QAbstractTableModel):
     def __init__(self):
         QAbstractTableModel.__init__(self)
         self.items = []
         self.bisector = None
+        self.is_downloading = False
```

Comment:
hm, could you set this on StepItem instead ? like the progress attribute. This way it may be easy for us when we will implement the background download. For now it should simplify the work of the delegate.